src/backend/cdb/dispatcher.c

```diff
@@ -1414,9 +1415,12 @@ dispatch_cleanup(DispatchData *data)
 	if (dispatcher_is_state_error(data))
```

Comment:
May we check the status of 'proc_exit_inprogress' earlier and simpler? That is:
```
if (proc_exit_inprogress)
    return;
```

Reply:
The same logic, no special performance improvement in my opinion